scene/resources/immediate_mesh.cpp

```diff
@@ -312,6 +312,8 @@ void ImmediateMesh::surface_end() {
 	uses_uv2s = false;
 
 	surface_active = false;
+
+	emit_changed();
```

Comment:
Was MeshInstance3D working previously? Or does this fix updating the AABB for MeshInstance3D as well?

Reply:
I'll have to try that.  I had only tried the mrp in the issue which used 2d meshes.

Reply:
To answer this question, no, I do not believe this happens with MeshInstance3d. That is because when  clear is called on the immediate mesh, that calls into the mesh storage which causes `mesh->dependency.changed_notify(Dependency::DEPENDENCY_CHANGED_MESH);` to get called. That flags the instance as dirty and on the next draw it updates its AABB so it stays up to date.

It doesn't seem the same Dependency tracking is used at all by the canvas items though.  The issue occurs because without the queue_redraw, the Rect2 that is used as bounds, doesn't get updated unless there is a draw command for the mesh (see https://github.com/godotengine/godot/blob/master/servers/rendering/renderer_canvas_render.cpp#L87).

It might be possible to add a DependencyTracker and use the same system for notifications for the canvas items as scene but where nothing else uses it I wonder if that is the right approach.  Array mesh also emits changed when you add a surface which allows anything higher to get nofiications via that means as well.  Which, like this change currently does, allows it to queue the redraw so the aabb for the canvas item gets updated(see https://github.com/godotengine/godot/blob/master/scene/resources/mesh.cpp#L1800)

Reply:
Thanks for taking a deeper look! I agree that your approach is the correct one for 2D. It's best to maintain the same approach as ArrayMesh